pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -49,13 +49,62 @@ const (
 
 	// defaultConfig is the default service config (that does not correspond to an actual service)
 	defaultConfig = "default"
+
+	// disablePolicy check
+	disablePolicy = "disablePolicy"
+
+	// enablePolicy check
+	enablePolicy = "enablePolicy"
+)
+
+var (
+	defaultdisableCheckPerFilterConfig = util.MessageToStruct(&mccpb.ServiceConfig{
+		DisableCheckCalls: true})
+	defaultenableCheckPerFilterConfig = util.MessageToStruct(&mccpb.ServiceConfig{
+		DisableCheckCalls: false})
 )
 
 // NewPlugin returns an ptr to an initialized mixer.Plugin.
 func NewPlugin() plugin.Plugin {
 	return mixerplugin{}
 }
 
+func (mixerplugin) OnPreComputePerRouteFilterConfig(env *model.Environment, push *model.PushContext) map[string]*types.Struct {
+	mixerPerRouteFilterConfig := make(map[string]*types.Struct)
+	for _, service := range push.Services(nil) {
+		// Use hostname for key
+		hostname := string(service.Hostname)
+		//calculate outbound
+		attrs := addDestinationServiceAttributes(make(attributes), push, service.Hostname)
+		config := &mccpb.ServiceConfig{
+			DisableCheckCalls: true,
+			MixerAttributes:   &mpb.Attributes{Attributes: attrs},
+			ForwardAttributes: &mpb.Attributes{Attributes: attrs},
+		}
+		filterConfig := util.MessageToStruct(config)
+		//key := string(service.Hostname) + "outbound" + disablePolicy
+		key := strings.Join([]string{hostname, "outbound", disablePolicy}, "|")
+		mixerPerRouteFilterConfig[key] = filterConfig
+		config = &mccpb.ServiceConfig{
+			DisableCheckCalls: false,
+			MixerAttributes:   &mpb.Attributes{Attributes: attrs},
+			ForwardAttributes: &mpb.Attributes{Attributes: attrs},
+		}
+		filterConfig = util.MessageToStruct(config)
+		key = strings.Join([]string{hostname, "outbound", enablePolicy}, "|")
```

Comment:
these flags are applicable only on the inbound path. Outbound does not have policy

Reply:
Confused. Why do you need two cache entries (with and without policy) in the outbound path? You need these on the inbound path only

Reply:
According to mixer.modifyOutboundRouteConfig:
	case *route.Route_Route:
		switch upstreams := action.Route.ClusterSpecifier.(type) {
		case *route.RouteAction_Cluster:
			_, _, hostname, _ := model.ParseSubsetKey(upstreams.Cluster)
			attrs := addDestinationServiceAttributes(make(attributes), push, hostname)
			httpRoute.PerFilterConfig = addServiceConfig(in.Node, httpRoute.PerFilterConfig, &mccpb.ServiceConfig{
				DisableCheckCalls: disableClientPolicyChecks(in.Env.Mesh, in.Node),
				MixerAttributes:   &mpb.Attributes{Attributes: attrs},
				ForwardAttributes: &mpb.Attributes{Attributes: attrs},
			})

There are two cases: outbound + with policy, outbound+without policy, so I put two cache entries here.